src/task_types/resmoke_config_writer.rs

```diff
@@ -351,16 +286,52 @@ impl ResmokeConfigActor for ResmokeConfigActorService {
     }
 }
 
+/// A cache for querying resmoke suite configurations.
```

Comment:
Calling out to resmoke tends to be really slow, so we want to minimize how frequently we do it. So this is just a simple cache so we only need to make the call out once (instead of for each sub-suite).